### PR TITLE
added integration test for question service

### LINKED ISSUE #27: Write integration test for question service

### PR DESCRIPTION
**Pull request by olehpona**
Created integretion test for question services.
The test consists of ten checks.

1. Should create question and return value 
2. Should find question by Id 
3. Should find question by author 
4. Should find question by title 
5. Should find question by category 
6. Should sort desc by updateData 
7. Should not update question when currentUser doesn't match author 
8. Should update question 
9. Should not delete question when currentUser doesn't match author 
10. Should delete question 

To run test run
```bash
npm test question.spec.js
```

### CODE REVIEW

**Review by vlad-khrychov — COMMENTED** [review]
I have not check all PR yet, but it is good example how to create DB "in memory", please share this experience with your team.

**Review by SachkovaYuliia — APPROVED** [review]

**Review by tufned — APPROVED** [review]
🔥

**Review by grbvtsk — APPROVED** [review]
nice work!

File: src/test/integration/services/question.spec.js [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+      id: question._id.toString(),
+      toUpdate: {
+        title: 'updateTest'
+      },
+      userId: '12345678'
+    }
+
+    expect(
+      async () => await questionService.updateQuestion(mockData.id, mockData.userId, mockData.toUpdate)
+    ).rejects.toThrow(createForbiddenError())
```
Comment (vlad-khrychov) [inline src/test/integration/services/question.spec.js:193]: In integration test we need to check the final result. In our case we can check that DB was not updated. For example maybe error generated but this update already is done.

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]
Add more end-result tests, not just function tests like in unit tests.

File: src/test/integration/services/question.spec.js (hunk 1 above)
Comment (olehpona) [reply src/test/integration/services/question.spec.js:193]: added verifacation that service don't call db update action

**Review by HalynaProlubshchykova — APPROVED** [review]

File: src/test/integration/services/question.spec.js (hunk 1 above)
Comment (vlad-khrychov) [reply src/test/integration/services/question.spec.js:193]: Can you also add `expect(questionMock.save).toHaveBeenCalledWith` in other test cases?

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**